test/metabase/query_processor/middleware/resolve_joins_test.clj

```diff
@@ -255,6 +255,35 @@
                                :fingerprint   nil}]}]
                    :limit  10}))))))
 
+(deftest ^:parallel native-model-field-ref-test
+  (testing "should use name-based field refs for joined native models with mapped database fields (metabase#58829)"
+    (let [source-metadata [{:id            (mt/id :checkins :id)
+                            :name          "_USER_ID"
+                            :display_name  "User ID"
+                            :base_type     :type/Integer
+                            :semantic_type :type/FK
+                            :field_ref     [:field "_USER_ID" {:base-type :type/Integer :join-alias "alias"}]
+                            :fingerprint   {:global {:distinct-count 15, :nil% 0.0}}}]]
+      (is (query= (mt/mbql-query users
+                    {:fields [$id
+                              [:field "_USER_ID" {:base-type :type/Integer :join-alias "alias"}]]
```

Comment:
nit: don't need to change this, but for the record, you can write this as `&alias.*_USER_ID/Integer`. 

The type can also be omitted if the main table (`users` here) has a field with the name after `*`, like `*LATITUDE`. The docs for this are on `metabase.test/$ids`.